internal/manifests/collector/servicemonitor.go

```diff
@@ -52,7 +52,8 @@ func ServiceMonitor(params manifests.Params) (*monitoringv1.ServiceMonitor, erro
 	name := naming.ServiceMonitor(params.OtelCol.Name)
 	labels := manifestutils.Labels(params.OtelCol.ObjectMeta, name, params.OtelCol.Spec.Image, ComponentOpenTelemetryCollector, []string{})
 	selectorLabels := manifestutils.SelectorLabels(params.OtelCol.ObjectMeta, ComponentOpenTelemetryCollector)
-	selectorLabels[monitoringLabel] = valueExists
+	// This label is to match only some services and exclude others like headless
+	selectorLabels[matchSMLabel] = "true"
```

Comment:
this is how we potentially create double targets which we should avoid.

Reply:
Why? I can change it but I want to understand.

Reply:
if service A and service B both point to the same ports on the same collector a servicemonitor that matches both services will generate two targets for the same collector pod

Reply:
yep, exactly right:

```mermaid
stateDiagram-v2
    classDef page fill:green
    classDef svc fill:blue

    ServiceMonitor --> collector_svc_monitoring
    ServiceMonitor  --> collector_svc
    collector_svc_monitoring --> collector_crd
    collector_svc --> collector_crd
    collector_crd --> collector_pod_1
    collector_crd --> collector_pod_2
    collector_crd --> collector_pod_3

    class collector_svc, collector_svc_monitoring svc
    class collector_pod_1, collector_pod_2, collector_pod_3 page
```